src/transport.rs

```diff
@@ -16,7 +16,20 @@ use crate::waiting_call_registry::WaitingCallRegistry;
 use crate::web_socket_connection::WebSocketConnection;
 use std::sync::mpsc::Receiver;
 
-pub type SessionId = String;
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct SessionId(String);
```

Comment:
can you tell me more about why this is preferable to a type alias? (I don't feel strongly either way, I'm just curious)

Reply:
When you type `pub type Kilometers = u64;` it doesn't make `Kilometers` a unique type. You can pass any `u64` to a function that requires `Kilometers` and if you also have `pub type Meters = u64;` then Rust won't warn you if you add kilometers to meters. 

I found this because one of the function required `&SessionId`, which is basically `&String`. Clippy complained that you should change that to `&str` (Like I changed `&Vec<t>` to `&[T]`), so I figured that it would be best to actually wrap session ID in a type.

Also, when you wrap types like I did it allows you not to derive traits that doesn't make sense. For example, if you have `struct UserId(u64)` then it makes sense to derive `PartialEq`, but doesn't make sense to derive `Add` because adding two user IDs together doesn't make any sense.